src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -361,6 +362,194 @@ namespace cryptonote
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  int t_cryptonote_protocol_handler<t_core>::handle_notify_new_fluffy_block(int command, NOTIFY_NEW_FLUFFY_BLOCK::request& arg, cryptonote_connection_context& context)
+  {
+    LOG_PRINT_CCONTEXT_L0("NOTIFY_NEW_FLUFFY_BLOCK (hop " << arg.hop << ")");
+    if(context.m_state == cryptonote_connection_context::state_normal)
+    {
+      m_core.pause_mine();
+      
+      block new_block;
+      transaction miner_tx;
+      if(parse_and_validate_block_from_blob(arg.b.block, new_block))
+      {
+        std::list<blobdata> have_tx;
+        std::list<crypto::hash> need_tx_hashes;
+        
+        transaction tx;
+        crypto::hash tx_hash;
+        std::unordered_map<crypto::hash, blobdata> tx_hash_to_blob;        
+        BOOST_FOREACH(auto& tx_blob, arg.b.txs)
+        {
+          if(parse_and_validate_tx_from_blob(tx_blob, tx))
+          {
+            cryptonote::tx_verification_context tvc = AUTO_VAL_INIT(tvc);
+            m_core.handle_incoming_tx(tx_blob, tvc, true, true);
+            if(tvc.m_verifivation_failed)
+            {
+              LOG_PRINT_CCONTEXT_L1("Block verification failed: transaction verification failed, dropping connection");
+              m_p2p->drop_connection(context);
+              m_core.resume_mine();
+              return 1;
+            }
+            
+            if(!tvc.m_added_to_pool) 
+            {
+              get_transaction_hash(tx, tx_hash);              
+              tx_hash_to_blob[tx_hash] = tx_blob;              
+            }
+          }
+          else
+          {
+            LOG_ERROR_CCONTEXT
+            (
+              "sent wrong tx: failed to parse and validate transaction: \r\n"
+              << epee::string_tools::buff_to_hex_nodelimer(tx_blob) 
+              << "\r\n dropping connection"
+            );
+            
+            m_p2p->drop_connection(context);
+            m_core.resume_mine();
+            return 1;
+          }
+        }
+        
+        BOOST_FOREACH(auto& tx_hash, new_block.tx_hashes)
+        {
+          if(m_core.get_pool_transaction(tx_hash, tx))
+          {
+              have_tx.push_back(tx_to_blob(tx));
+          }
+          else
+          {
+            if(tx_hash_to_blob.find(tx_hash) != tx_hash_to_blob.end()) 
+            {
+                have_tx.push_back(tx_hash_to_blob[tx_hash]);
```

Comment:
So the only way it wouldn't be added to the pool is if it's a bad transaction?
If it was rejected wouldn't it fail the tvc.m_verifivation_failed check earlier?

I'm almost certain I added that because I was getting tripped up by transactions passing the check, but not being added to the pool. Could have something to do with me setting do_not_relay to true in wallet2.cpp for testing?